libnopegl/src/block.c

```diff
@@ -228,6 +228,19 @@ void ngli_block_field_copy(const struct block_field *fi, uint8_t *dst, const uin
     ngli_block_field_copy_count(fi, dst, src, 0);
 }
 
+void ngli_block_fields_copy(const struct block *s, const struct block_field_data *src_array, uint8_t *dst)
+{
+    const struct block_field *fields = ngli_darray_data(&s->fields);
+    for (size_t i = 0; i < ngli_darray_count(&s->fields); i++) {
+        const struct block_field_data *src = &src_array[i];
+        const struct block_field *fi = &fields[i];
+        if (src->count)
+            ngli_block_field_copy_count(fi, dst + fi->offset, src->data, src->count);
+        else
+            ngli_block_field_copy(fi, dst + fi->offset, src->data);
```

Comment:
I think you can use unconditionnally ngli_block_field_copy_count() here to avoid another branch in ngli_block_field_copy.

Reply:
Ah good point, changed